cpp/dolfinx/fem/Expression.h

```diff
@@ -112,11 +114,39 @@ class Expression
     // Create data structures used in evaluation
     std::vector<double> coordinate_dofs(3 * num_dofs_g);
 
+    int num_argument_dofs = 1;
+    xtl::span<const std::uint32_t> cell_info;
+    std::function<void(const xtl::span<T>&,
+                       const xtl::span<const std::uint32_t>&, std::int32_t,
+                       int)> dof_transform
+        = [](const xtl::span<T>&, const xtl::span<const std::uint32_t>&,
+             std::int32_t, int)
+    {
+      // Do nothing
+    };
+
+    if (_argument_function_space)
+    {
+      num_argument_dofs
+          = _argument_function_space->dofmap()->element_dof_layout().num_dofs();
+      const auto element = _argument_function_space->element();
+
+      if (element->needs_dof_transformations())
+      {
+        _mesh->topology_mutable().create_entity_permutations();
+        cell_info = xtl::span(_mesh->topology().get_cell_permutation_info());
+        dof_transform = element->get_dof_transformation_function<T>();
```

Comment:
@michalhabera On my new version of clang this gave me an error unless I removed the <T>?